deploy/base/deployment.yaml

```diff
@@ -0,0 +1,64 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: objectstorage-provisioner
+  namespace: objectstorage-provisioner-ns
+  labels:
+    app.kubernetes.io/part-of: container-object-storage-interface
+    app.kubernetes.io/component: provisioner
+    app.kubernetes.io/version: main
+    app.kubernetes.io/name: container-object-storage-interface-provisioner-sidecar
+spec:
+  replicas: 1
+  minReadySeconds: 30
+  progressDeadlineSeconds: 600
+  revisionHistoryLimit: 3
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+  selector:
+    matchLabels:
+      app.kubernetes.io/part-of: container-object-storage-interface
+      app.kubernetes.io/component: provisioner
+      app.kubernetes.io/name: container-object-storage-interface-provisioner-sidecar
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/part-of: container-object-storage-interface
+        app.kubernetes.io/component: provisioner
+        app.kubernetes.io/name: container-object-storage-interface-provisioner-sidecar
+    spec:
+      serviceAccountName: objectstorage-provisioner-sa
+      containers:
+      - name: objectstorage-sample-driver
+        image: quay.io/containerobjectstorage/objectstorage-sample-driver
+        imagePullPolicy: IfNotPresent
+        resources:
+          limits:
+            cpu: 1
+            memory: 1Gi
+          requests:
+            cpu: 100m
+            memory: 100Mi
+        envFrom:
+        - secretRef:
+            name: objectstorage-provisioner
+      - name: objectstorage-provisioner-sidecar
+        image: quay.io/containerobjectstorage/objectstorage-provisioner-sidecar
+        imagePullPolicy: IfNotPresent
+        resources:
+          limits:
+            cpu: 1
+            memory: 1Gi
+          requests:
+            cpu: 100m
+            memory: 100Mi
+        env:
+        - name: CONNECT_ADDRESS
+          valueFrom:
+            secretKeyRef:
+              name: objectstorage-provisioner
+              key: LISTEN_ADDRESS
```

Comment:
Not sure a listen address should be kept in a secret. It's not really secret, right?